code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1298 @@
+	else
+		..()
+
+/obj/structure/mannequin/cyber/examine(mob/user)
+	..()
+	if(destroyed)
+		to_chat(user, "Its glass shield has been shattered.")
+
+/obj/structure/mannequin/cyber/broken
+	health = 100
+	destroyed = 1
+
+
+
+
+
+/////////////////////////////////////////////////////////MANNEQUIN FRAME//////////////////////////////////////////////////////////
+//Used to build cyber mannequins.
+
+/obj/structure/mannequin_frame
+	name = "human cyber mannequin frame"
+	desc = "Lots of work just to display some clothes."
+	icon = 'icons/obj/mannequin_64x64.dmi'
+	icon_state="mannequin_cyber_human_frame"
+	pixel_x = -16
+	anchored = 0
+	density = 0
+	var/datum/construction/construct
+
+/obj/structure/mannequin_frame/New()
+	..()
+	construct = new /datum/construction/mannequin_frame(src)
+
+/obj/structure/mannequin_frame/attackby(var/obj/item/W, var/mob/user)
+	if(!construct || !construct.action(W, user))
+		..()
+
+/datum/construction/mannequin_frame/custom_action(step, atom/used_atom, mob/user)
+	user.visible_message("[user] has connected [used_atom] to [holder].", "You connect [used_atom] to [holder]")
+	holder.overlays += image(holder.icon, used_atom.icon_state)
+	qdel (used_atom)
+	used_atom = null
+	return 1
+
+/datum/construction/mannequin_frame/action(atom/used_atom,mob/user as mob)
+	return check_all_steps(used_atom,user)
+
+/datum/construction/mannequin_frame
+	steps = list(
+				list(Co_KEY=/obj/item/robot_parts/head),
+				list(Co_KEY=/obj/item/robot_parts/chest),
+				list(Co_KEY=/obj/item/robot_parts/r_leg),
+				list(Co_KEY=/obj/item/robot_parts/l_leg),
+				list(Co_KEY=/obj/item/robot_parts/r_arm),
+				list(Co_KEY=/obj/item/robot_parts/l_arm)
+				)
+
+/datum/construction/mannequin_frame/spawn_result(mob/user as mob)
+	var/obj/structure/mannequin_frame/const_holder = holder
+	const_holder.construct = new /datum/construction/reversible/mannequin(const_holder)
+	const_holder.overlays.len = 0
+	const_holder.icon_state = "mannequin_cyber_human"
+	const_holder.overlays |= icon(const_holder.icon, "lightout")
+	spawn()
+		qdel (src)
+
+/datum/construction/reversible/mannequin
+
+/datum/construction/reversible/mannequin
+	result = /obj/structure/mannequin/cyber
+	var/base_icon = "station_map_frame"
+
+	steps = list(
+		list(
+			Co_DESC="The circuitboard is installed",
+			Co_NEXTSTEP = list(
+				Co_KEY=/obj/item/stack/sheet/glass/glass,
+				Co_AMOUNT = 1,
+				Co_VIS_MSG = "{USER} install{s} the glass shield to {HOLDER}.",
+				Co_DELAY = 20
+				),
+			Co_BACKSTEP = list(
+				Co_KEY=/obj/item/weapon/crowbar,
+				Co_VIS_MSG = "{USER} prie{s} the circuitboard from {HOLDER}."
+				)
+			),
+		list(
+			Co_DESC="The frame is on the wall.",
+			Co_NEXTSTEP = list(
+				Co_KEY=/obj/item/weapon/circuitboard/airlock,
+				Co_AMOUNT = 1,
+				Co_VIS_MSG = "{USER} install{s} the circuitboard into {HOLDER}.",
+				Co_AMOUNT = 1
+				),
+			Co_BACKSTEP = list(
```

Comment:
You can go back and forth through the steps, that's the whole point. It'll be more useful after we (eventually) add multiple options per step for stuff like mechs and spacepods with custom plates.

Optional steps were giving me a lot of grief to implement, but I might go back to them.